### PR TITLE
[MAINTENANCE] Delete usage_statistics_handler from the set of RuleBasedProfilerResult and DataAssistantResult object serialization keys

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Delete `DataAssistantResult._usage_statistics_handler` and `RuleBasedProfilerResult._usage_statistics_handler` `to_dict()` serialization keys (it is deemed not useful in these objects).
* Ensure that existing tests pass.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1101
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py [hunk 1]
```diff
@@ -1647,6 +1647,7 @@ def test_volume_data_assistant_result_serialization(
         bobby_volume_data_assistant_result.to_json_dict()
         == volume_data_assistant_result_as_dict
     )
+    assert len(bobby_volume_data_assistant_result.profiler_config.rules) == 2
```
Comment (cdkini) [inline tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py:1650]: Why is this here? I don't think it fits the scope of the PR right?

**Review by cdkini — APPROVED** [review]

File: tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py (hunk 1 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py:1650]: @cdkini Good catch -- I felt to add this as a nice test for the symmetry with the onboarding data assistant test of the same kind.  Thanks!

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM :)
